master/buildbot/data/builds.py

```diff
@@ -53,8 +72,18 @@ def get(self, resultSpec, kwargs):
             bldr = kwargs['builderid']
             num = kwargs['number']
             dbdict = yield self.master.db.builds.getBuildByNumber(bldr, num)
-        defer.returnValue((yield self.db2data(dbdict))
-                          if dbdict else None)
+        data = yield self.db2data(dbdict) if dbdict else None
+        filters = resultSpec.popProperties() if hasattr(resultSpec, 'popProperties') else []
+        # Avoid to request DB for Build's properties if not specified
+        if filters:  # pragma: no cover
+            try:
+                props = yield self.master.db.builds.getBuildProperties(data['buildid'])
+            except (KeyError, TypeError):
+                props = {}
+            filtered_properties = self._generate_filtered_properties(props, filters)
+            if filtered_properties:
+                data['properties'] = filtered_properties
```

Comment:
is it possible that data == `None` (as per line 75) and we try to add a value to a non-dictionary?